tests/parser/test_class2D.py

```diff
@@ -47,53 +56,80 @@ def test_all_keys_are_different(input):
 def test_counter(input):
     class2d_object = input
     class_numbers = class2d_object.class_number
-    class2d_object._count_all(class_numbers)
+    for i in range(len(class_numbers)):
+        class2d_object._count_all(class_numbers[i])
 
 
 def test_percentage(input):
     class2d_object = input
     class_numbers = class2d_object.class_number
-    percentage = class2d_object.percent_all_particles_per_class(class_numbers)
-    print("Percent of particles from all data in each class:", percentage)
-    print(
-        "Percent of all particles included in the 20 most populated classes:",
-        sum(x[1] for x in percentage),
-    )
-    assert percentage[0][1] == 18.327067669172934
+    job_numbers = class2d_object.job_number
+    percentage = None
+    for i in range(len(job_numbers)):
+        percentage = class2d_object.percent_all_particles_per_class(class_numbers[i])
+        print("Percent of particles from all data in each class:", percentage)
+        print(
+            "Percent of all particles included in the 20 most populated classes for",
+            job_numbers[i],
+            ":",
+            sum(x[1] for x in percentage),
+            "%",
+        )
+    assert percentage[0][1] == 19.9430499894537
 
 
 def test_top_twenty_list(input):
     class2d_object = input
     class_numbers = class2d_object.class_number
-    twenty_list = class2d_object.top_twenty_most_populated(class_numbers)
-    print("20 most populated classes:", twenty_list)
+    twenty_list = []
+    for i in range(len(class_numbers)):
+        twenty_list = class2d_object.top_twenty_most_populated(class_numbers[i])
```

Comment:
this looks to me like we're not actually interested in `i`, but only want to iterate over the dictionary values.
```suggestion
    for class in class_numbers.values():
        twenty_list = class2d_object.top_twenty_most_populated(class)
```
(I'm assuming `class_numbers` is a dictionary, but it could also be a list, in which case you can just loop over the list)